lib/instana/secrets.rb

```diff
@@ -0,0 +1,41 @@
+require 'uri'
+require 'cgi'
+
+module Instana
+  class Secrets    
+    def remove_from_query(str, secret_values = Instana.agent.secret_values)
+      return str unless secret_values
+      
+      url = URI(str)
+      params = CGI.parse(url.query)
+      
+      redacted = params.map do |k, v|
+        needs_redaction = secret_values['list']
+          .any? { |t| matcher(secret_values['matcher']).(t,k) }
+        [k, needs_redaction ? '<redacted>' : v]
+      end
+      
+      url.query = URI.encode_www_form(redacted)
+      CGI.unescape(url.to_s)
+    end
+    
+    private
+    
+    def matcher(name)
+      case name
+      when 'equals-ignore-case'
+        ->(expected, actual) { expected.casecmp(actual) == 0 }
+      when 'equals'
+        ->(expected, actual) { (expected <=> actual) == 0 }
+      when 'contains-ignore-case'
+        ->(expected, actual) { actual.downcase.include?(expected) }
+      when 'contains'
+        ->(expected, actual) { actual.include?(expected) }
+      when 'regex'
+        ->(expected, actual) { !Regexp.new(expected).match(actual).nil? }
```

Comment:
Since we're not interested in match results, wouldn't `match?` be enough here?

```suggestion
        ->(expected, actual) { Regexp.new(expected).match?(actual) }
```

Reply:
`Regexp#match?` was added in [2.4.0](https://github.com/ruby/ruby/blob/48b94b791997881929c739c64f95ac30f3fd0bb9/doc/ChangeLog-2.4.0#L4739-L4742). Our current minimum supported version is [2.1.0](https://www.instana.com/docs/ecosystem/ruby/supported-versions/).